"cannot convert argument 2 from 'char **' to 'const char **'" under MSVC

Under MSVC, https://github.com/libofx/libofx/commit/9023c0f8b7744baebb28e2ff9b9e6e54ed0a4d33 causes MSVC to fail with:

```
 error C2664: 'size_t iconv(iconv_t,const char **,size_t *,char **,size_t *)': cannot convert argument 2 from 'char **' to 'const char **'
```

This isn't an issue under vcpkg, though, likely due to some leniency stemming from a particular set of compiler switches.

--- a/lib/ofx_preproc.cpp
+++ b/lib/ofx_preproc.cpp
@@ -308,8 +308,12 @@ int ofx_proc_file(LibofxContextPtr ctx, const char * p_filename)
             const char* inchar = s_buffer.c_str();
             char * outchar = iconv_buffer;
             int iconv_retval = iconv (conversion_descriptor,
-                                      const_cast<char**>(&inchar), &inbytesleft,
-                                      &outchar, &outbytesleft);
+#ifdef _WIN32 //Win32's iconv implementation takes a const char** inchar
+                                      &inchar,
+#else
+                                      const_cast<char**>(&inchar),
+#endif
+                                      &inbytesleft, &outchar, &outbytesleft);
             if (iconv_retval == -1)
             {
               message_out(ERROR, "ofx_proc_file(): Iconv conversion error");

